CtrlVAF/CtrlVAF.Core/Core/Dispatcher.cs

```diff
@@ -84,5 +90,31 @@ public TReturn Dispatch(Action<Exception> exceptionHandler, params ICtrlVAFComma
         /// <returns>The result or <see cref="default"/> for no result.</returns>
         protected internal abstract TReturn HandleConcreteTypes(IEnumerable<Type> types, params ICtrlVAFCommand[] commands);
 
+        public virtual Dispatcher<TReturn> ClearCache()
+        {
+            TypeCache = new ConcurrentDictionary<Type, IEnumerable<Type>>();
+            ResultsCache = new ConcurrentDictionary<Type, TReturn>();
+            return this;
+        }
+
+        /// <summary>
+        /// Gets the cached results for the key type.
+        /// </summary>
+        /// <param name="keyType"></param>
+        /// <returns></returns>
+        public TReturn GetCachedResults(Type keyType)
+        {
+            return ResultsCache[keyType];
```

Comment:
To ensure that what you want to get is actually there, otherwise default
```suggestion
            if(!ResultsCache.TryGetValue(keyType, out var value)) {
                return default;
            }
            return value;
```